airbyte-integrations/connectors/source-survey-sparrow/source_survey_sparrow/survey_sparrow.yaml

```diff
@@ -0,0 +1,111 @@
+version: "0.1.0"
+
+definitions:
+  page_size: 50
+  selector:
+    extractor:
+      field_pointer: ["data"]
+  requester:
+    url_base: "https://eu-api.surveysparrow.com/v3"
+    http_method: "GET"
+    authenticator:
+      type: BearerAuthenticator
+      api_token: "{{ config['access_token'] }}"
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: "DefaultPaginator"
+      page_size_option:
+        inject_into: "request_parameter"
+        field_name: "limit"
+      pagination_strategy:
+        type: "PageIncrement"
+        page_size: "*ref(definitions.page_size)"
+      page_token_option:
+        inject_into: "request_parameter"
+        field_name: "page"
+      url_base: "*ref(definitions.requester.url_base)"
+    requester:
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  contacts_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "contacts"
+      primary_key: "id"
+      path: "/contacts"
+  contact_lists_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "contact_lists"
+      primary_key: "id"
+      path: "/contact_lists"
+    paginator:
+      type: "NoPagination"
+  questions_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "questions"
+      primary_key: "id"
+      path: "/questions"
+    retriever:
+      $ref: "*ref(definitions.base_stream.retriever)"
+      requester:
+        $ref: "*ref(definitions.requester)"
+        request_options_provider:
+          request_parameters:
+            survey_id: "{{ config['survey_id'] }}"
```

Comment:
This will break if you don't inform the survey_id.

Options:
```python
config.get('survey_id')
```

Reply:
@girarda would be possible to create an array in the spec.yaml and consume as a list and send multiple requests?

https://developers.surveysparrow.com/rest-apis/questions#getV3Questions The endpoint uses only one survey, my idea is give the option to retrieve an array and get more data :D

Reply:
@marcosmarxm, this is a great idea!

That can be done using a [ListStreamSlicer](https://docs.airbyte.com/connector-development/config-based/understanding-the-yaml-file/stream-slicers#list-stream-slicer).

The values can refer values from the spec and you can use `inject_into: "request_parameter"` to update the request params

Reply:
@jhajajaas what do you think about this suggestion?

Reply:
@girarda @marcosmarxm I've tried implementing `ListStreamSlicer` - here's an example for one stream:

```yaml
  questions_stream:
    $ref: "*ref(definitions.base_stream)"
    $options:
      name: "questions"
      primary_key: "id"
      path: "/questions"
    retriever:
      $ref: "*ref(definitions.base_stream.retriever)"
      stream_slicer:
        type: "ListStreamSlicer"
        slice_values: "{{ config.get('survey_id') }}"
        cursor_field: "survey_id"
        request_option:
          field_name: "survey_id"
          inject_into: "request_parameter"
```

This causes an error since the `survey_id` is a required parameter. From what I tested, if we have survey_id = [1, 2] airbyte will make 3 requests: first with no survey_id and second and third with survey_id = 1 and 2 respectively. Any workaround on this?

Reply:
@girarda it works if I use hardcoded values:
```yaml
  questions_stream:
    $ref: "*ref(definitions.base_stream)"
    $options:
      name: "questions"
      primary_key: "id"
      path: "/questions"
    retriever:
      $ref: "*ref(definitions.base_stream.retriever)"
      stream_slicer:
        type: "ListStreamSlicer"
        slice_values:  # removed this one "{{ config.get('survey_id') }}"
           - 123
           - 345
        cursor_field: "survey_id"
        request_option:
          field_name: "survey_id"
          inject_into: "request_parameter"
```

Using the config way the request is built like: `{"type": "DEBUG", "message": "Making outbound API request", "data": {"headers": "{'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive', 'Authorization': 'Bearer ****'}", "request_body": "None", "url": "https://api.surveysparrow.com/v3/questions?limit=10"}}`

Reply:
for me it's still not working even though I hardcoded the slice_values (I'm using cdk v4.0.1)

Reply:
@jhajajaas @girarda found the possible solution to the bug. Let's wait until get this solved. Besides that your connector looks great!

Reply:
@jhajajaas can you try Airbyte CDK 0.4.2, Alex merged a fix to your issue: https://github.com/airbytehq/airbyte/pull/18574/files